### DISCUSSION TITLE
Better devtools experience

### DISCUSSION THREAD

**Discussion by bmacher** [vuejs/pinia#519] [Ideas]
Before opening a feature request right away I wanted to ask if there is the possibility to somehow improve the devtools experience. Instead of saying something like `[store] direct` it would be much better if the invoked action would be shown. Another point is the payload of the mutations. There is one when using `$patch()` inside the action. But since there are no real mutations anymore, couldn't you just list the params (as array) of invoked the action?

To summarize:

`user.someAction(123, 'str')`

[user] someAction

mutation
  payload: [123, 'str']
  type: someAction

**Comment by posva** [comment]
What version of pinia? Can tou explain with screenshots? You can create an issue for this 🙂

**Reply by bmacher** [reply]
I am currently using `^0.5.1` and therefore v1.

This is what I see:

![image](https://user-images.githubusercontent.com/32685580/120166763-3d0bc400-c1fd-11eb-8c75-e71186baa2d5.png)

![image](https://user-images.githubusercontent.com/32685580/120166814-4b59e000-c1fd-11eb-8e8c-7eb32884f0b1.png)

What I would like to see in the first case:

`[station| fetchAll` &rarr; which is the action name

`payload: []` &rarr; because it was invoked with no params

What I would like to see in the second case:

`[notification| showNotification` &rarr; which is the action name

`payload: ["Station erfolgreich gelöscht.", "success", true]`

**Reply by posva** [reply]
Ah I see, thanks for the clear examples. Yeah, this isn't possible for that version of devtools. I will probably deprecate and drop support for the old devtools now that the beta of vue devtools support Vue 2. It's a big refactoring work though (from v2) so I will likely address other things first like documentation examples

**Reply by bmacher** [reply]
So should I still open an issue?

**Reply by posva** [reply]
Created https://github.com/posva/pinia/issues/521 🙂

**Reply by bmacher** [reply]
Thanks
